### DISCUSSION TITLE
Cannot Connect to ESP32 in AP Mode

### DISCUSSION THREAD

**Discussion by vonCorax** [micropython/micropython#11293] [Using MicroPython]
I've run into a problem setting my ESP32-WROOM-32 to run in AP mode, where my code works with MicroPython v1.17 but not with v1.19.1. (It also fails with v1.18, although I haven't tested this extensively.) My code is as follows:
```
try:
    import usocket as socket
except:
    import socket

import network

import esp
esp.osdebug(None)

import gc
gc.collect()

ssidAP         = 'WiFi_ESP32'
passwordAP     = '12345678'

ap_if = network.WLAN(network.AP_IF)

def AP_Setup(ssidAP, passwordAP):
    print("Setting soft-AP  ... ")
    ap_if.active(True)
    ap_if.config(essid=ssidAP,authmode=network.AUTH_WPA_WPA2_PSK, password=passwordAP)
    while not ap_if.active():
        pass
    print('Success, IP address:', ap_if.ifconfig())
    print("Setup End\n")

try:
    AP_Setup(ssidAP, passwordAP)
except:
    print("Failed, please disconnect the power and restart the operation.")
    ap_if.disconnect()
```

When I run this code with v1.17 I can connect to the AP using my Android phone (Android 12), but when I try with v1.19.1 my phone spins for about a minute and then reports that it was unable to obtain an IP address. Is this a reportable bug, or am I doing something wrong?

**Comment by davefes** [comment]
I never use AP mode, however for STA l always do the following:
```py
sta_if = network.WLAN(network.STA_IF)

def connect():
      sta_if.active(True)

 #  seems more reliable to start with a fresh connect()
    if sta_if.isconnected():
        sta_if.disconnect()
        print ('started in the connected state, but now disconnected')
    else:
        print ('started in the disconnected state')

    utime.sleep(0.1)
```
Try doing the same with your AP interface and then add the above.

Or try this:
https://github.com/glenn20/micropython-espnow-utils/blob/main/src/wifi.py
I use it on all my ESPNow/WiFi projects.
